### ISSUE TITLE
More visibility for route direction

### ISSUE THREAD

**Issue by Stefal** [osmandapp/OsmAnd#9975]
### Description

When your route passes several times on the same road, the map becomes confusing:

![Screenshot_20201007-164908](https://user-images.githubusercontent.com/8255984/95348965-db3dc900-08be-11eb-99e5-d253e401bd72.png)

### Describe the solution you'd like

Perhaps OsmAnd could hide or add some transparency on the route that we'll follow much later.

**Comment by scaidermern** [comment]
Maybe it would be sufficient to introduce a casing for turn arrows and to draw them in reverse order, i.e. from latest turn to earliest turn. This could result in a visualization similar to:

![95348965-db3dc900-08be-11eb-99e5-d253e401bd72](https://user-images.githubusercontent.com/1040784/95420639-15997b80-093c-11eb-88df-1525e5acb83d.png)

**Comment by Stefal** [comment]
That could be a solution, we need to check if "detecting" the right arrow on the screen doesn't need too much attention.

**Comment by Stefal** [comment]
Any update about this ? I tried to use gradient as a workaround but it doesn't work.

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/8255984/192152361-9291d083-4348-41a6-a2f2-6b7a1f1d952f.jpg"  width="500" height="auto"/> </td>
  </tr>
</table>

**Comment by pyrog** [comment]
> Maybe it would be sufficient to introduce a casing for turn arrows and to draw them in reverse order

I suggest to draw only the next arrow.

**Comment by pyrog** [comment]
Also, arrows on the track are difficult to read when tracks are superposed.
I suggest to draw them only on i.e.  100 m on front of the car, or only on the "top part" of the track.

<img width="156" src="https://user-images.githubusercontent.com/2158081/192158661-f361481b-1684-4b08-b595-4f6c6626ab25.png"> <img width="164" alt="Capture d’écran 2022-09-25 à 20 09 14" src="https://user-images.githubusercontent.com/2158081/192158667-4d333eab-a579-4010-8157-0854ceb0e9a6.png">

**Comment by pyrog** [comment]
Note: the usage of these tracks is to take pictures of the "all" roads of a city.

**Comment by Stefal** [comment]
@vshcherb What can we do to move this issue upper on the todo list? 
Pro Subscription is enabled for me. I will ask my 2 collegues to do the same thing

**Comment by vshcherb** [comment]
Yes it's confusing we need to redesign how navigation works so it should draw one style line to the first point and other style to others. This is simply not a proper use case for current navigation and I don't think it's possible to solve it with just line rendering

**Comment by Davi427** [comment]
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1d08d1f0-921d-4174-8c47-8a744c7fb451"  width="300" height="auto"/> </td>
  </tr>
</table>

Another example of the current rendering making it not particularly clear where to go, when a route goes over the same place multiple times.
Especially with a large roundabout like the one here,the 50kmh speed limit and the large number of exits (the unhelpful navigation instruction showing the 6th exit as being a right turn when it isn't), the route line would be quite helpful to select the correct exit. Current rendering makes that a challenge. 

Arrows
If only the current arrow would be drawn in bright yellow and all the future arrows in some other, less visible colour then it would already be a major improvement while keeping the possibility of zooming out to see the other parts of the route.

Route line 
Can we not use the colour/gradient system that is already used in the custom route lines here?
The first, idk, 200 metres are the bright colour chosen for the route line (in my example pink) and after those 200m it fades the saturation to be very low. That way there is a clear difference between current and future lines where they overlap.

**Comment by scaidermern** [comment]
> and after those 200m it fades the saturation to be very low

If the route line fades, it will make it hard to get an overview of the entire route.

**Comment by Davi427** [comment]
> > and after those 200m it fades the saturation to be very low
> 
> If the route line fades, it will make it hard to get an overview of the entire route.

I'm pretty sure there is a level of saturation that balances short term visibility and overview while also accounting for people with colour vision issues.
Ideally it would be a configurable option in the route line menu so that everyone can set it to their own preference instead of a centralised decision having to be "good enough" for all users.

**Comment by seen-one** [comment]
https://github.com/user-attachments/assets/2b403088-837e-451d-a3b2-7e20802efa35

I managed to AI generate a hacky patch, but it would be a very nice option to have!

**Comment by Rowin63** [comment]
> > > Maybe it would be sufficient to introduce a casing for turn arrows and to draw them in reverse order
> > 
> > 
> > I suggest to draw only the next arrow.
> 
> This is another possibility. However, sometimes you want to pan the map and take a look ahead of the next turns. Drawing only the next turn arrow will even increase confusion about these next turns if they overlap but don't have visible turn arrows.

Solving this, really would be an important improvement. 
Perhaps highlight just the upcoming turn arrow (yellow as usual) and grey out the following ones?
